src/server/src/test/java/io/cassandrareaper/acceptance/ReaperTestJettyRunner.java

```diff
@@ -80,267 +76,22 @@ public SimpleReaperClient getClient() {
-        return Class.forName(REAPER_APPLICATION_FQN);
-      } catch (MalformedURLException | ClassNotFoundException ex) {
-        throw new AssertionError(ex);
-      }
-    }
-
-    public int getLocalPort() {
-      try {
-        // reflection for `dropwizardTestSupport.getLocalPort()`
-        return (Integer)supportCls.getMethod("getLocalPort").invoke(support);
-      } catch (ReflectiveOperationException ex) {
-        throw new AssertionError(ex);
-      }
-    }
-
-    public void before() {
-      try {
-        // reflection for `dropwizardTestSupport.before()`
-        supportCls.getMethod("before").invoke(support);
-      } catch (ReflectiveOperationException ex) {
-        throw new AssertionError(ex);
-      }
-    }
-
-    public void after() {
-      try {
-        Class<?> type;
-        Method method;
-        Object context = getContext();
-        // reflection for `context.schedulingManager.cancel()`
-        Field field = context.getClass().getField("schedulingManager");
-        {
-          Object schedulingManager = field.get(context);
-          if (null != schedulingManager) {
-            type = field.getType();
-            method = type.getMethod("cancel");
-            method.invoke(schedulingManager);
-          }
-        }
-        {
-          // reflection for `context.repairManager.close()`
-          field = context.getClass().getField("repairManager");
-          Object repairManager = field.get(context);
-          if (null != repairManager) {
-            type = field.getType();
-            method = type.getDeclaredMethod("close");
-            method.invoke(repairManager);
-          }
-        }
-        {
-          // reflection for `context.isRunning.set(false)`
-          field = context.getClass().getField("isRunning");
-          type = field.getType();
-          method = type.getDeclaredMethod("set", Boolean.TYPE);
-          method.invoke(field.get(context), false);
-        }
-        try {
-          Thread.sleep(100);
-          // reflection for `dropwizardTestSupport.after()`
-          supportCls.getMethod("after").invoke(support);
-        } catch (InterruptedException expected) { }
-      } catch (ReflectiveOperationException ex) {
-        throw new AssertionError(ex);
-      }
-    }
-
-    String getContextStorageClassname() {
-      try {
-        // reflection for `context.storage.getClass().getName()`
-        Object context = getContext();
-        return context.getClass().getField("storage").get(context).getClass().getName();
-      } catch (ReflectiveOperationException ex) {
-        throw new AssertionError(ex);
-      }
-    }
-
-    private static int getAnyAvailablePort() {
-      try (ServerSocket s = new ServerSocket(0)) {
-        return USED_PORTS.add(s.getLocalPort()) ? s.getLocalPort() : getAnyAvailablePort();
-      } catch (IOException ex) {
-        throw new IllegalStateException("no available ports", ex);
-      }
-    }
+  public AppContext getContext() {
```

Comment:
`getContext()` now returns `AppContext` directly instead of `Object`. This has a side-effect on all call sites: the cast to `AppContext` is now useless and should be removed.

Reply:
I removed all casts 👍